server/app/repository/ProgramRepository.java

```diff
@@ -173,6 +173,32 @@ public CompletableFuture<Program> getActiveProgramFromSlug(String slug) {
         executionContext.current());
   }
 
+  /** Get the current draft program with the provided slug. */
+  public CompletableFuture<Program> getDraftProgramFromSlug(String slug) {
+    return supplyAsync(
+        () -> {
+          for (Program program : database.find(Program.class).where().isNull("slug").findList()) {
+            program.getSlug();
+            program.save();
+          }
+
+          Optional<Version> version = versionRepository.get().getDraftVersion();
+
+          if (version.isEmpty()) {
+            throw new RuntimeException(new ProgramNotFoundException(slug));
```

Comment:
**suggestion (blocking):** 

1. would prefer this throws a checked exception so the caller decides how to handle it rather than the repository deciding for the caller that this will cause a server error.

2. use a different exception type -- `ProgramNotFoundException` doesn't match the exceptional case here that there is no draft version.

Reply:
For (1), this is tricky because Java does not allow checked exceptions in CompletableFutures, even if we declare `throws ProgramNotFoundException` in the method signature (since the method itself isn't throwing it). That's probably why the existing `getActiveProgramFromSlug` above is wrapping it in a `RuntimeException` as well.

For this I could make this method synchronous (avoiding the supplyAsync block) since there isn't any async code calling it right now, or we could leave the unchecked exception as-is. What do you think?

For (2), I could still make the underlying `ProgramNotFoundException` a different one, perhaps a new `ProgramDraftNotAvailableException`?

Reply:
> For this I could make this method synchronous (avoiding the supplyAsync block) since there isn't any async code calling it right now, or we could leave the unchecked exception as-is. What do you think?

I'd prefer this -- better to avoid the complexity of async when we don't need it.

Reply:
For (2): the important thing is to communicate to callers what could go wrong and what actually went wrong when it goes wrong. The problem wasn't that they asked for a program that doesn't exist, it's that they should not call this method when no draft version exists, so the exception and javadoc should reflect that.

Reply:
Made `getDraftProgramFromSlug` and `getDraftProgramDefinition` synchronous. Created and used new `ProgramDraftNotFoundException`.